### PR TITLE
feat(category): make category inmemory api capable of taking page_siz…

### PR DESCRIPTION
**Pull request by lderrickable**
…e and current_page

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The category inmemory api can't correctly process a request with a page_size or current_page.

Fixes: N/A

## What is the new behavior?
The category inmemory api correctly processes requests with a page_size or current_page.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/category/testing/src/drivers/in-memory/category.service.ts [hunk 1]
```diff
@@ -13,6 +13,11 @@ export class DaffInMemoryCategoryService implements DaffCategoryServiceInterface
   constructor(private http: HttpClient) {}
 
   get(categoryRequest: DaffCategoryRequest): Observable<DaffGetCategoryResponse> {
-    return this.http.get<DaffGetCategoryResponse>(this.url + categoryRequest.id);
+		const params = new HttpParams()
+			.set('page_size', categoryRequest.page_size ? categoryRequest.page_size.toString() : null)
+			.set('current_page', categoryRequest.current_page ? categoryRequest.current_page.toString() : null)
+			.set('id', categoryRequest.id);
```
Comment (damienwebdev) [inline libs/category/testing/src/drivers/in-memory/category.service.ts]: Why do you pass an `id` param? Isn't this already url param?

File: libs/category/testing/src/inmemory-backend/category.service.ts [hunk 2]
```diff
@@ -55,5 +59,19 @@ export class DaffInMemoryBackendCategoryService implements InMemoryDbService {
...
       };
     });
-  }
+	}
+
+	private generatePageSize(reqInfo) {
+		if(reqInfo.req.params.map && reqInfo.req.params.map.get('page_size') && reqInfo.req.params.map.get('page_size')[0]) {
+			return parseInt(reqInfo.req.params.map.get('page_size')[0], 10);
+		}
+		return Math.floor(Math.random() * 8 + 1);
```
Comment (damienwebdev) [inline libs/category/testing/src/inmemory-backend/category.service.ts]: Can this be a constant?

File: libs/category/testing/src/drivers/in-memory/category.service.ts (hunk 1 above)
Comment (lderrickable) [reply libs/category/testing/src/drivers/in-memory/category.service.ts]: removed

File: libs/category/testing/src/inmemory-backend/category.service.ts (hunk 2 above)
Comment (lderrickable) [reply libs/category/testing/src/inmemory-backend/category.service.ts]: I made it be 10 unless the user passes one in

**Pull request merged**